neonvm/runner/main.go

```diff
@@ -1137,17 +1137,50 @@ func defaultNetwork(logger *zap.Logger, cidr string, ports []vmv1.Port) (mac.MAC
 	}
 
 	// pass incoming traffic to .Guest.Spec.Ports into VM
+	iptablesArgs := []string{}
 	for _, port := range ports {
-		logger.Info(fmt.Sprintf("setup DNAT for incoming traffic, port %d", port.Port))
-		iptablesArgs := []string{
+		logger.Info(fmt.Sprintf("setup DNAT rule for incoming traffic to port %d", port.Port))
+		iptablesArgs = []string{
 			"-t", "nat", "-A", "PREROUTING",
 			"-i", "eth0", "-p", fmt.Sprint(port.Protocol), "--dport", fmt.Sprint(port.Port),
 			"-j", "DNAT", "--to", fmt.Sprintf("%s:%d", ipVm.String(), port.Port),
 		}
 		if err := execFg("iptables", iptablesArgs...); err != nil {
-			logger.Error("could not set up DNAT for incoming traffic", zap.Error(err))
+			logger.Error("could not set up DNAT rule  for incoming traffic", zap.Error(err))
 			return nil, err
 		}
+		logger.Info(fmt.Sprintf("setup DNAT rule for traffic originating from localhost to port %d", port.Port))
+		iptablesArgs = []string{
+			"-t", "nat", "-A", "OUTPUT",
+			"-m", "addrtype", "--src-type", "LOCAL", "--dst-type", "LOCAL",
+			"-p", fmt.Sprint(port.Protocol), "--dport", fmt.Sprint(port.Port),
+			"-j", "DNAT", "--to-destination", fmt.Sprintf("%s:%d", ipVm.String(), port.Port),
+		}
+		if err := execFg("iptables", iptablesArgs...); err != nil {
+			logger.Error("could not set up DNAT rule for traffic from localhost", zap.Error(err))
+			return nil, err
+		}
+		logger.Info(fmt.Sprintf("setup ACCEPT rule for traffic originating from localhost to port %d", port.Port))
+		iptablesArgs = []string{
```

Comment:
Why do we need the accept rule here?

Reply:
it's  OUTPUT chain needed because in neon-compute we have additional iptables rules that prevent any traffic to 169.254.252.252/30